File: Lab2/Task2/ControlExperiment/app/src/main/java/com/example/controlexperiment/MainActivity.java
package com.example.controlexperiment;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.Spinner;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
// Initialize UI elements
        EditText editText = findViewById(R.id.editText);
        Button button = findViewById(R.id.button);
        CheckBox checkBox = findViewById(R.id.checkBox);
        RadioGroup radioGroup = findViewById(R.id.radioGroup);
        Spinner spinner = findViewById(R.id.spinner);
// Set up spinner data
        String[] options = {"Option A", "Option B", "Option C"};
        ArrayAdapter<String> adapter = new ArrayAdapter<>(this,
                android.R.layout.simple_spinner_item, options);
        adapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        spinner.setAdapter(adapter);
// Button click listener
        button.setOnClickListener(v -> {
            String input = editText.getText().toString().trim();

            //Validate EditText input
            if(input.isEmpty()){
                editText.setError("Field cannot be empty");
                return;
            }

            boolean isChecked = checkBox.isChecked();
            int selectedId = radioGroup.getCheckedRadioButtonId();
            RadioButton selectedRadioButton =
                    findViewById(selectedId);
            String selectedRadioText = selectedRadioButton != null ?
                    selectedRadioButton.getText().toString() : "None";
            String selectedSpinner =
                    spinner.getSelectedItem().toString();
// Show a Toast message with the collected input
            Toast.makeText(this, "Input: " + input + "\nChecked: " +
                            isChecked +
                            "\nSelected Radio: " + selectedRadioText +
                            "\nSelected Spinner: " + selectedSpinner,
                    Toast.LENGTH_LONG).show();
        });
    }
}

File: Lab3/task4/app/src/main/java/com/example/fragmentexample/MainActivity.kt
package com.example.fragmentexample

import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
class MainActivity : AppCompatActivity(), MyInterface {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val fragment1 = Fragment1()
        val fragment2 = Fragment2()
        supportFragmentManager.beginTransaction().add(R.id.fl_fragment1, fragment1, "Fragment1").commit()
        supportFragmentManager.beginTransaction().add(R.id.fl_fragment2, fragment2, "Fragment2").commit()
    }
    override fun setResult(data: String) {
        val fragment2 = supportFragmentManager.findFragmentByTag("Fragment2") as Fragment2
        fragment2.setResult(data)
    }
}